flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SynchronousCheckpointITCase.java

```diff
@@ -108,6 +110,9 @@ public void initializeLatchesAndError() {
 		error.set(null);
 	}
 
+	@Rule
+	public final Timeout timeoutPerTest = Timeout.seconds(7);
```

Comment:
Why did you pick 7 seconds? I think usually we set the timeouts to 30, 60 or 120 seconds. 7 seconds might be too fast on travis.

Reply:
Doesn't it depend on a test? I see a lot of tests in Flink using 10, 5, 2 seconds as timeout values, it's hard to tell that 30, 60 or 120 is a "usual" value.

On my dev machine, the test runs takes about less than 2.5 second. 7 seconds is about x3 of the estimated run time. On some machines 5 seconds may be too little, but more than 10 seconds is too much for this test, imo.

Do you have some other value in mind?

Reply:
Just saying from my experience: I would set the timeout to 60 seconds.

Travis can often stall for many seconds. I've never investigated the reasons why (swapping? gc? vm failover?), but the problem with travis is not that it's 2 or 3 times slower, but that it can randomly stall from time to time. For example a test might usually execute in 2 seconds, but from time to time it will take 30 seconds to complete.

On the other hand, is there any benefit of timeouting those tests in 7s instead of 60s? On travis I think there is none. Locally there is benefit only if you are debugging this particular test failure for shorter feedback loop, which is also a rare case and one can manually and temporarily lower the timeout. I think tests that are setting some lower timeouts are actively waiting on it and that creates a motivation to lower the timeout as low as possible (to speed up the tests).

(If you insist on trying this 7s timeout, I won't block it)

Reply:
I don't insist on 7 seconds at all, and as mentioned before, I'm open for other values.

I just think that with too big timeouts (like 30 seconds and above), it's more probable that something wrong with the test itself, rather with the environment running it (**note:** we are discussing only this concrete test, "YMMV" for other tests). In that case I'd prefer the test to fail reasonably fast (even on CI, as it uses shared resources).

> LGTM % timeout. If you don't want to change it, let me know and I will merge it,

If you want, **I can change it to 10 seconds before merging this PR**. Otherwise, it looks like we are both not enforcing any specific value, so we can leave it as is for now.

> but in that case please monitor jira issues if someone is reporting this test as flaky.

I'll adjust the timeout (new or current) if needed and if I notice such issue, but I don't promise to vigorously monitor jira issues for that.